tools/bin/hawqregister

```diff
@@ -1145,7 +1153,7 @@ def main(options, args):
 
     failure_handler = FailureHandler(conn)
     # register
-    if options.yml_config and ispartition(options.yml_config):
+    if check_file_exist(options.yml_config) and ispartition(options.yml_config):
```

Comment:
It will always check yaml file exist or not first.  Then check whether is partition. if partition do HawqRegisterPartition, otherwise HawqRegister. So the check works  before doing register partition table or non-partition table. So there is no need to call the function 2 times.